apple/MarkdownTextInputDecoratorComponentView.mm

```diff
@@ -173,6 +203,12 @@ - (void)applyNewStyles
   }
 }
 
+- (void)prepareForRecycle
+{
+  [super prepareForRecycle];
+  _markdownUtils = [[RCTMarkdownUtils alloc] init];
```

Comment:
If I understand correctly, the implementation of `prepareForRecycle` should be somewhat identical to `initWithFrame`. Do we need to reset `_props` and `_observersAdded` here as well?

Reply:
I don't think we need to reset props since on the `Insert` mutation, `updateProps` will be called with the `oldProps` being a `nullptr`: https://github.com/facebook/react-native/blob/d631ec9a6c3cd2b5321de9ee35553878f1428b05/packages/react-native/React/Fabric/Mounting/RCTMountingManager.mm#L73-L91. I added it to be safe though.

When it comes to `_observersAdded` - no. It should never be true at this point, I've added an assert to reflect that.